### PR TITLE
Update getBlock and getTransaction to Support Versioned Transactions

### PR DESCRIPTION
**Pull request by enzoampil**
This pull request implements versioned transaction support for the `get_block` and `get_transaction` methods ( issue #295  ) by allowing users to pass `max_supported_transaction_version` when calling these functions, which are then used for the corresponding RPC configs from `solders`: `RpcBlockConfig ` and `RpcTransactionConfig` .

I've run the recommended [dev setup steps](https://github.com/michaelhly/solana-py#-development) including `make format` and `make lint`, which explains why there are some formatting changes included in the PR.

Happy to make changes based on any feedback!

### CODE REVIEW

File: src/solana/rpc/api.py
Comment (roks0n) [inline src/solana/rpc/api.py:399]: Missing a mention that it's an optional kwarg?

File: src/solana/rpc/async_api.py
Comment (roks0n) [inline src/solana/rpc/async_api.py:409]: same here (sorry for nit)

**Review by roks0n — COMMENTED** [review]
2 nits, but other than that it looks good! Adding my review as a comment as I'm not sure how strict the maintainer is regarding docstrs. :)

Appreciate the PR!

**Review by kevinheavey — APPROVED** [review]

**Review by roks0n — APPROVED** [review]

**Comment by a26nine** [comment]
@kevinheavey, when would the release (with this commit) come out?

**Comment by a26nine** [comment]
@michaelhly @kevinheavey Any ETA on when this update would be pushed out?

**Comment by kevinheavey** [comment]
Planning for a release after #302 gets merged

**Comment by a26nine** [comment]
#302 seems to be merged now... 😶

**Pull request merged**
